helm-service/controller/deployment_handler.go

```diff
@@ -154,6 +154,27 @@ func (h *DeploymentHandler) catchupGeneratedChartOnboarding(deploymentStrategy k
 	return h.onboarder.OnboardGeneratedChart(userChartManifest, event, deploymentStrategy)
 }
 
+func (h *DeploymentHandler) getPortOfDeployedService(e keptnv2.EventData) (string, error) {
+	userChartManifest, err := h.getHelmExecutor().GetManifest(helm.GetReleaseName(e.Project, e.Stage, e.Service, false),
+		e.Project+"-"+e.Stage)
+
+	if err != nil {
+		return "", err
+	}
+	services := helm.GetServices(userChartManifest)
+	if len(services) > 0 {
+		if len(services[0].Spec.Ports) > 0 {
+			for _, port := range services[0].Spec.Ports {
+				if port.Protocol == corev1.ProtocolTCP {
+					return fmt.Sprintf("%d", port.Port), nil
+				}
+			}
+		}
+
+	}
+	return "80", nil
```

Comment:
Can we add some logging here, stating that no services could be found and that we are defaulting to port 80?

Reply:
Actually, thinking about this: If no (valid) services could be found, would it make sense to raise an error?

Reply:
Yes, raising an error would be the right approach I think. Although in this case (i.e. when no port is defined), it will not reach that code anyway since the deployment will fail - K8s Services always require a port to be defined

Reply:
Then let's raise an error here, just in case, so we can see if this happens.

Reply:
I have changed the logic to raise an error if a service has been defined, but no port could be found. Further, I changed the code to only generate a deploymentURI if a service has actually been defined in the helm chart.

Reply:
That sounds nice, thanks!